consensus/ethash/api.go

```diff
@@ -61,6 +64,18 @@ func (api *API) GetWork() ([3]string, error) {
 	}
 }
 
+// GetWork returns a default work package for external miner.
+//
+// Note this function is a legacy for backward compatibility.
+func (api *API) GetWork() ([3]string, error) {
+	return api.getWork("")
+}
+
+// GetCustomizedWork returns a customized work package for external miner.
+func (api *API) GetCustomizedWork(extra string) ([3]string, error) {
+	return api.getWork(extra)
+}
```

Comment:
You don't need a second enpoind. You can make `func (api *API) GetWork(extra *string) ([3]string, error) {`. If the user doesn't supply an `extra`, you interpret it one way, if they do supply `extra`, you interpret it the other way.